deepface/modules/recognition.py

```diff
@@ -3,6 +3,7 @@
 import pickle
 from typing import List, Union, Optional, Dict, Any
 import time
+import imghdr
```

Comment:
I discourage the use of `imghdr` as already shows "deprecated" and support will be removed soon. Maybe PIL is better

Reply:
Good spot

Reply:
Recommended way to determine file type with PIL is

```python
from PIL import Image
img = Image.open(filename)
```

But I do not want to load and open file. If this is the way, then I will prefer to use file extension.

Reply:
You have to read carefully the PIL documentation for the open method on image:

> This is a lazy operation; this function identifies the file, but   the file remains open and the actual image data is not read from the file until you try to process the data (or call the :py:meth:`~PIL.Image.Image.load` method).  

Also img.format reads only as many bytes needed to parse the header (pretty much like imghder.what) hence the operation is very cheap.

Reply:
Cheers!